server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/hotspot/HotspotApi.java

```diff
@@ -182,14 +182,13 @@ private static ServerHotspot adapt(Hotspots.SearchWsResponse.Hotspot hotspot, St
       filePath,
       convertTextRange(hotspot.getTextRange()),
       ServerApiUtils.parseOffsetDateTime(hotspot.getCreationDate()).toInstant(),
-      getStatus(hotspot),
-      VulnerabilityProbability.valueOf(hotspot.getVulnerabilityProbability()));
+      getStatus(hotspot.getStatus(), hotspot.getResolution()),
+      VulnerabilityProbability.valueOf(hotspot.getVulnerabilityProbability()),
+      hotspot.getAssignee());
   }
 
-  private static HotspotReviewStatus getStatus(Hotspots.SearchWsResponse.Hotspot hotspot) {
-    var status = hotspot.getStatus();
-    if ("REVIEWED".equals(status) && hotspot.hasResolution()) {
-      var resolution = hotspot.getResolution();
```

Comment:
There is a specificity with protobuf, generated getters never return null for Strings. The default value is the empty string `""`. The presence or absence of a field is normally checked with `hasXXX` methods. So here the second part of your new condition will always evaluate as `true`. I don't think it's an issue, if the payload is formed as expected, if the status is REVIEWED there should always be a resolution. The previous check was a kind of guard just in case